### DISCUSSION TITLE
Anomally with Points

### DISCUSSION THREAD

**Discussion by Warren-Whitfield** [pelicanmapping/osgearth#2498] [Support Q&A]
Why when I do this...

osg::ref_ptr<MultiGeometry> multi = new MultiGeometry();
osg::ref_ptr<LineString> pline = new LineString(2);
pline->push_back(m_BARLine.gpStart.x(), m_BARLine.gpStart.y(), m_BARLine.gpStart.z());
pline->push_back(m_BARLine.gpStop.x(), m_BARLine.gpStop.y(), m_BARLine.gpStop.z());

osg::ref_ptr<Point> basepoint = new Point(1);
basepoint->set(osg::Vec3d(m_BARLine.gpStart.x(), m_BARLine.gpStart.y(), m_BARLine.gpStart.z()));

multi->add(basepoint);
multi->add(pline);

Does the point appear on both ends of the line like this? 
![image](https://github.com/gwaldron/osgearth/assets/159110675/b8be913c-07f4-4bb8-874d-fc34501dc72f)

**Comment by gwaldron** [comment]
Probably b/c you have a Style that includes a point symbol. It will apply to the entire geometry including the line segment.

**Comment by Warren-Whitfield** [comment]
So I have these styles defined...
//Create a style for the lines, colour, fill etc..
m_Style.getOrCreate<LineSymbol>()->stroke()->color() = m_lineColour;
m_Style.getOrCreate<LineSymbol>()->stroke()->width() = m_lineWidth;
m_Style.getOrCreate<LineSymbol>()->stroke()->smooth() = true;
m_Style.getOrCreate<LineSymbol>()->stroke()->stipple() = m_lineStipple;
m_Style.getOrCreate<PolygonSymbol>()->fill()->color() = m_fillColour;
m_Style.getOrCreate<PointSymbol>()->fill()->color() = m_fillColour;
m_Style.getOrCreate<PointSymbol>()->smooth() = true;
m_Style.getOrCreate<AltitudeSymbol>()->clamping() = AltitudeSymbol::CLAMP_NONE;
m_Style.getOrCreate<TextSymbol>()->fill()->color() = m_textColour;
m_Style.getOrCreate<TextSymbol>()->size() = m_fontSize;
m_Style.getOrCreate<TextSymbol>()->font() = m_fontName;

...but if I comment out the PointSymbol then I lose the ability to colour and size and it becomes just a white square, still on both ends though?
Alternatively, is there another way to add line endings? Typically I'd want different endings for start and end sides. For example, a circle and an arrow.

**Reply by gwaldron** [reply]
It applies all your symbology to the geometry at once. If you don't want that, you will probably have to create separate geometries each with a distinct style.

Points render as squares (or a circle if you use smoothing). If you want a custom shape you might try an IconSymbol instead.
